apps/search/src/components/AlgoliaConfigurationForm.tsx

```diff
@@ -56,15 +53,12 @@ export const AlgoliaConfigurationForm = () => {
     });
 
   const onFormSubmit = handleSubmit(async (conf) => {
-    const client = new AlgoliaSearchProvider({
-      appId: conf.appId ?? "",
-      apiKey: conf.secretKey ?? "",
-      indexNamePrefix: conf.indexNamePrefix,
-      enabledKeys: [], // not required for ping but should be refactored
-    });
-
     try {
-      await client.ping();
+      await algoliaCredentialsVerifier.verifyCredentials({
+        appId: conf.appId,
+        apiKey: conf.secretKey,
+      });
```

Comment:
extracted to sepraate class to improve app architecture. We don't have to create entire client with entire config, to validate credentials. I could removed a lot of code thanks to that